pkg/overlay/overlay_linux.go

```diff
@@ -63,6 +63,13 @@ func MountWithOptions(contentDir, source, dest string, opts *Options) (mount spe
 			if err := os.Chown(upperDir, int(stat.Uid), int(stat.Gid)); err != nil {
 				return mount, err
 			}
+			times := []syscall.Timeval{
```

Comment:
Why out of curiosity? Is this helping ensure reproducibility? We aren't serializing anything related to the timestamps of this directory to e.g. a tar stream are we?

Reply:
Without it, two different runs would appear to have different timestamps on the top level of the build context directory, breaking one of the tests added by #5691.